operator-wg/whitepaper-community-review/README.md

```diff
@@ -0,0 +1,948 @@
+
+These capabilities should be transparent to the users the user should have the possibility to override such auto-configuration mechanisms if they want to do so. Furthermore, automatic reconfigurations should be well-documented in a way that the user could comprehend what is happening on the infrastructure.
+
+#### Uninstalling / Disconnect
+When deleting the declarative requested state (in most cases a custom resource), an operator should allow two behaviors:
+- Uninstalling: An operator should be able to completely remove or delete every managed resource.
+- Disconnecting: An operator should stop managing the provisioned resources.
+
+Both processes should be applied to every resource that the operator directly provisioned.  
+An operator should report any failure in the process in a declarative way (using the [status field](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status) for example).
+
+## Security
+![operator model](img/04_1_operator_model.png)
+
+Operators are intended to manage their state and configuration via
+the Kubernetes API server using the Custom Resource Definition. The
+subordinate API resources they manage (often pods running stateful
+applications) also have their lifecycle and supporting RBAC, services,
+etc. managed via the Kubernetes API. In some cases the operator
+will also interact with the application’s API across the network.
+All of these routes offer potential to compromise the operator and
+its resources, and should be protected in line with best practices
+laid out below.
+
+### Operator Developer
+Operator developers should be aware of the security risks an operator
+introduces and document its secure use. While developing an operator
+it's important to focus on key areas such as transparency and
+documentation, operator scope, and vulnerability analysis.
+
+#### Transparency and Documentation
+During development of an operator a developer should have a clear understanding
+of how it will work and interface within Kubernetes. As a developer shifts
+from development to publishing the operator, users should also
+have a clear understanding of what the operator does, and how.
+
+You've written something you're proud of, but think of this from
+the end user's point of view: Should they trust source code from
+the internet, an operator to run with administrative access on their
+cluster which may be large and costly, or may be handling sensitive
+information?  Anything the developer can do to help a user come up
+to speed with their software, how it works, how it's secured, and
+what affects it might have on their cluster will make it easier for
+them to adopt the software.
```

Comment:
```suggestion
During the development of an operator, a developer should have a clear understanding of how it will work and interface within Kubernetes. As developers shift from development to publishing the operator, users should also have a clear understanding of what the operator does, and how.
You've written something you're proud of, but think of this from the end user's point of view: Should they trust source code from the internet, an operator to run with administrative access on their cluster which may be large and costly, or maybe handling sensitive information? Anything the developer can do to help a user come up to speed with their software, how it works, how it's secured, and what effects it might have on their cluster will make it easier for them to adopt the software.
```